### ISSUE TITLE
Tests may hang

### ISSUE THREAD

**Issue by eserte** [kentnl/IO-Async-XMLStream-SAXReader#2]
I observe on several of my smoker systems hanging tests (with 0.001002, but I saw it also with older versions). My FreeBSD smokers are not affected, only some of my Linux smokers. Please let me know if you need more information (strace logs, perl -d:Trace output, whatever).

**Comment by kentfredric** [comment]
Yeah. I'm aware of this, seen issues in reports. But haven't been able to get a handle on what's going on due to inability to reproduce.

I *assume* this situation is what is causing "waitstat: 9" ( #1 ) 

Working with IO Loops and testing is hard :(

**Comment by kentfredric** [comment]
I'll probably need to summon a @Leonerd and hopefully he can tell me what I've done wrong in my [tests](https://github.com/kentnl/IO-Async-XMLStream-SAXReader/blob/master/t/01-piecewise-xml.t)

But I think the only conclusion I can make is it never reaches EOF and instead waits forever.

**Comment by leonerd** [comment]
Oh hi. I realise I'd missed this earlier. Is this still an issue? Do you still need help here?
